routing.md

```diff
@@ -305,12 +321,51 @@ DIDComm DID Document endpoints have the following format:
 
 **type**: MUST be `DIDComm`. 
 
-**serviceEndpoint**: MUST contain a URI for a transport specified in the [transports] section of this spec.
+**serviceEndpoint**: MUST contain a URI for a transport specified in the [transports] section of this spec, or a URI from Alternative Endpoints.
 
 **routingKeys**: An optional ordered array of strings referencing keys to be used when preparing the message for transmission as specified in the [Routing] section of this spec. 
 
 #### Multiple Endpoints
 
 A DID Document may contain multiple service entries of type `DIDComm`. Unless otherwise specified, messages should be sent to endpoints in the order they are specified.
 
+#### Alternative Endpoints
+
+In addition to the URIs for [transports], some alternative forms are available.
+
+##### DID
+
+Using a DID for the `serviceEndpoint` is useful when using a mediator. The DID should be resolved, and services with type of "DIDComm" will contain valid `serviceEndpoints`. The keyAgreement keys of that DID Document should be appended at the end of the routingKeys section from the message recipient's DID Document as per the process in [Sender Forward Process]. The key advantage with this approach is that a mediator can rotate keys and update serviceEndpoints without any updates needed to dependent recipients` DID Documents.
+
+A DID used as a mediator in this way MUST NOT use alternative endpoints. Only URIs described in [transports] are acceptable.
+
+Example 1: Mediator
+
+```json
+{
+    "id": "did:example:123456789abcdefghi#didcomm-1",
+    "type": "DIDComm",
+    "serviceEndpoint": "did:example:somemediator"
```

Comment:
This will need to be a didUrl rather than a did, so the specific service endpoint can be de-referenced properly.

Reply:
does it? What if any of the transports are valid? As the goal is to get the message to them, I don't think that locking it to one specific transport is helpful.

Reply:
serviceEndpoint would need to be a URL is what I'm saying. did:example:123 is a URI did:example:123#endpoint1 is a URL.

See [here](https://w3c.github.io/did-core/#did-syntax) for the URI syntax and [here](https://w3c.github.io/did-core/#did-url-syntax) for the did URL syntax

Reply:
The DID Core Spec indicates that it should be a URI. We could further constrain that for the DIDComm use case, but allowing a DID (URI) has a very useful application here and is allowable.

So, I disagree that serviceEndpoint needs to be a URL.

Reply:
Agree now that I understand how it's being used. We'll need to further define expected behavior of uri resolution. (E.g. MUST select a service endpoint from the resolved did doc with type: DIDComm, SHOULD NOT resolve deeper then two redirects)

Reply:
Line 340 already contains the MUST NOT recurse (use of alternative endpoints).
Line 351 already contains the requirement for the mediator endpoint to be of type `DIDComm`.

Does the language of either need to be adjusted?

Reply:
I did adjust line 340 to use SHOULD NOT instead of MUST NOT.

Reply:
Thanks, I made one minor suggestion to the wording because I wasn't certain I understood what you were trying to say. In general, I think in general, we've addressed my concerns though and anything additional is editorial remarks.